src/sysfiles.c

```diff
@@ -339,6 +340,18 @@ static Obj FuncCrcString(Obj self, Obj str)
     return INTOBJ_INT(((Int4) crc) >> 4);
 }
 
+// Get OS Kernel version. Used to discover if GAP is running inside
+// 'Windows Subystem for Linux'
+Obj SyGetOsRelease(void)
+{
+    struct utsname buf;
+    if (!uname(&buf)) {
```

Comment:
If you wrap uname, how about going the full way and return a record with all components? And then just return `fail` if there is an error?

Reply:
(to be clear, this is not a change required for this PR to be merged. You might want to apply the suggestion by Wilf, though)